src/NuGet.Core/NuGet.Packaging/Core/PackageDependencyInfo.cs

```diff
@@ -18,7 +18,12 @@ namespace NuGet.Packaging.Core
     /// </remarks>
     public class PackageDependencyInfo : PackageIdentity, IEquatable<PackageDependencyInfo>
     {
+#if NET45
```

Comment:
I was not sure why this preprocessor directive has been added. Later I found that [Array.Empty<T> Method](https://docs.microsoft.com/en-us/dotnet/api/system.array.empty?view=net-5.0) is introduced in .NET Framework 4.6 and higher versions.

Reply:
Yeah, it's kind of edge case. Up until this I didn't know we have some project with net 4.5 😄 
So far I have seen mostly net 4.7, netstandard 2.1, net 5.0 projects.